crates/ra_db/src/input.rs

```diff
@@ -177,6 +179,15 @@ impl CrateGraph {
         self.arena[&crate_id].edition
     }
 
+    /// Returns a name of a crate, declared in the root project.
+    /// May be missing for some cases, such as when the crate definition was created for a code snippet.
+    ///
+    /// This should not be considered as a normal crate name, since the actual name can be different in
+    /// a particular dependent crate, where it is specified.
+    pub fn declaration_name(&self, crate_id: &CrateId) -> Option<&String> {
```

Comment:
```suggestion
    pub fn declaration_name(&self, crate_id: &CrateId) -> Option<&str> {
```

`&String` is one of the "useless" types in Rust, along with `&Vec<T>`, `&Box<T>`, etc. 

Also, I feel we should redo this API to try to be less abstract, and just expose CrateData:

```
crate_graph[crate_id].deps()
```

But this is unrelated to the PR.

Reply:
>  we should redo this API to try to be less abstract, and just expose CrateData

That was easy to do, so why not.